src/thunderbots/software/ai/navigator/obstacle/obstacle.h

```diff
@@ -3,27 +3,77 @@
+#include "geom/angle.h"
 #include "geom/point.h"
 #include "geom/polygon.h"
 #include "geom/util.h"
+#include "shared/constants.h"
 
 class Obstacle
 {
    public:
+    static Obstacle createRobotObstacle(const Robot& robot, bool enable_velocity_cushion);
+
     /*
      * Gets the boundary polygon around the given robot obstacle that other robots
-     * should not enter with a buffer scaled by radiusScaling and
-     * velocity_projection_scaling
+     * should not enter with a buffer scaled by radius_cushion_scaling and
+     * velocity_cushion_scaling
+     *
+     * The radius cushion is a hexagon such that the centre to side distance is (minimum
+     * distance needed for two robots to clear each other * radius scaling)
+     *
+     * The velocity cushion is a rectangular projection whose length is (velocity *
+     * velocity scaling) and whose width is the diameter of the radius cushion
+     *
      *
      * @param robot robot to create obstacle boundary polygon around
-     * @param radiusScaling safety factor to vary the radius size of the buffer zone
-     * (1=default), must be greater than 0
-     * @param velocity_projection_scaling scales the projection buffer in the direction of
-     * the velocity (1=default), must be greater than 0
+     * @param radius_cushion_scaling multiplicatively scales the radius of the obstacle
+     * @param velocity_cushion_scaling multiplicatively scales the cushion of
+     * velocity of the robot
      *
      * @return a six-sided Polygon to represent the boundary around the obstacle
      */
-    static Polygon getBoundaryPolygon(const Robot& robot, double robot_radius_scaling,
-                                      double velocity_projection_scaling);
+    static Obstacle createRobotObstacleWithScalingParams(const Robot& robot,
+                                                         double radius_cushion_scaling,
+                                                         double velocity_cushion_scaling);
+
+    /*
+     * Gets the boundary polygon around the given robot obstacle that other robots
+     * should not enter with a buffer additively expanded by
+     * additional_radius_cushion_buffer and additional_velocity_cushion_buffer
+     *
+     * The radius cushion is a hexagon such that the centre to side distance is (minimum
+     * distance needed for two robots to clear each other + radius buffer)
+     *
+     * The velocity cushion is a rectangular projection whose length is (velocity +
+     * velocity buffer) and whose width is the diameter of the radius cushion
+     *
+     * @param robot robot to create obstacle boundary polygon around
+     * @param additional_radius_cushion_buffer additively increases the radius of the
+     * obstacle by specified amount
+     * @param additional_velocity_cushion_buffer additively increases the cushion of
+     * velocity of the robot
+     * @param enable_velocity_cushion if true, enables the cushion of the velocity
```

Comment:
Similar comment with this boolean parameter to above?

Reply:
With the buffer param because it's a fixed value, you can't turn off the velocity cushion (unless you knew what the velocity was), so this parameter is necessary.